### PR TITLE
EIP 7702 Implement SetCode transactions

### PR DESCRIPTION
**Pull request by anvacaru**
Add EIP-7702 SetCode Transaction Type Implementation

Adding support for a new transaction type that can include signed authorizations from various accounts. Each authorization lets you set a special code  (`0xEF0100 || address`) for said authorized account, which points to another account/SC. Later, whenever anyone interacts with the authorized account, instead of just sending funds, the system runs the contract code that was pointed to (`address`). 

Changes include:

#### Key Components:
- Add `SetCode` transaction type with prefix `0x04`
- Define `SetCodeTx` data structure with authorization list
- Add processing for `SetCode` authority lists
- Implement the delegation indicator (0xEF0100) for EOAs
- Add gas refund mechanism for existing accounts
- Delegation Resolution: When calling an account, check if its code represents a valid delegation using `#isValidDelegation`, and fetch the code of the delegated account.

#### Notable Changes:
- Add `#loadAuthorities` and `#addAuthority` operations for processing
- Implement authority recovery from message parameters
- Add verification of authority chain ID, address, and nonce
- Update the `Cextra` function when computing the cost of a call to also check if there is a delegation and if the delegated account is warm. 
- Support account delegation and code clearing

#### TODO:
- ~Need to implement delegation resolution in `CALL*`/`operations~
- ~Run conformance tests~

### CODE REVIEW

File: kevm-pyk/src/tests/utils.py [hunk 1]
```diff
@@ -40,6 +40,7 @@ def _assert_exit_code_zero(pattern: Pattern) -> None:
         return
 
     pretty = kore_print(pattern, definition_dir=kdist.get('evm-semantics.llvm'), output=PrintOutput.PRETTY)
+    print(pretty)
```
Comment (palinatolmach) [inline kevm-pyk/src/tests/utils.py:43]: Is this intended?

File: kevm-pyk/src/kevm_pyk/kproj/evm-semantics/driver.md [hunk 2]
```diff
@@ -117,7 +117,7 @@ To do so, we'll extend sort `JSON` with some EVM specific syntax, and provide a
     syntax EthereumCommand ::= loadTx ( Account ) [symbol(loadTx)]
  // --------------------------------------------------------------
     rule <k> loadTx(_) => startTx ... </k>
-         <statusCode> _ => EVMC_OUT_OF_GAS </statusCode>
```
Comment (palinatolmach) [inline kevm-pyk/src/kevm_pyk/kproj/evm-semantics/driver.md:120]: This change looks good, but I think this tule was removed in https://github.com/runtimeverification/evm-semantics/pull/2756/files. Which one is the latest version, I think the EIP-7623's one?

File: tests/execution-spec-tests/failing.llvm [hunk 3]
```diff
@@ -1794,7 +1788,48 @@ blockchain_tests/prague/eip7623_increase_calldata_cost/transaction_validity/tran
...
 blockchain_tests/prague/eip7623_increase_calldata_cost/transaction_validity/transaction_validity_type_3.json,tests/prague/eip7623_increase_calldata_cost/test_transaction_validity.py::test_transaction_validity_type_3[fork_Prague-blockchain_test_from_state_test-type_3-single_blob-single_access_list_single_storage_key-insufficient_gas-floor_gas_greater_than_intrinsic_gas]
-blockchain_tests/prague/eip7623_increase_calldata_cost/transaction_validity/transaction_validity_type_4.json,*
+blockchain_tests/prague/eip7623_increase_calldata_cost/transaction_validity/transaction_validity_type_4.json,tests/prague/eip7623_increase_calldata_cost/test_transaction_validity.py::test_transaction_validity_type_4[fork_Prague-blockchain_test_from_state_test-type_4-multiple_authorizations-multiple_access_lists_multiple_storage_keys-exact_gas-floor_gas_greater_than_intrinsic_gas]
```
Comment (palinatolmach) [inline tests/execution-spec-tests/failing.llvm:1791]: Do you know why these are failing?

Comment (anvacaru) [reply tests/execution-spec-tests/failing.llvm]: I missed some edge cases that prevented tests from passing. For example, [here](https://github.com/runtimeverification/evm-semantics/pull/2755/commits/2c67e8006a2b304e8d0a49aa51cc94f507eab924#diff-5567ecde96d0960be69f6f3f6353fd9d5e72d6662261a4991098057cebe421c4R803), I did not ensure that the width of the `sigR` and `sigS` is 32, which is required by [ECDSARecover](https://github.com/runtimeverification/blockchain-k-plugin/blob/master/plugin-c/crypto.cpp#L117) hook used by `#sender`.

There are still two tests for EIP7702 that seem to fail due to a JSON parsing error, I plan to address these in a follow-up PR.
```
pattern = App(symbol="Lbl'Hash'parseJSONs2List'LParUndsRParUnds'STATE-UTILS'Unds'List'Unds'JSONs", sorts=(), args=(App(symbol='L...ra'", sorts=(), args=()))))))))))))))),)), App(symbol="Lbl'Stop'List'LBraQuot'JSONs'QuotRBra'", sorts=(), args=()))),))
```
